### PR TITLE
Add game state management for Tic-tac-toe poker.

### PR DESCRIPTION
**Pull request by chuanyu**

### CODE REVIEW

File: src/rules/tictactoepoker/tictactoepoker.ts [hunk 1]
```diff
@@ -0,0 +1,752 @@
...
+    }
+}
+
+export interface ScoreWithExplanations {
+    score: number,
+    explanations: ScoreExplanation[]
+}
+
+// Return score type for the given three cards (some of all of which may be empty cards)
+function compute_score(a: Card, b: Card, c: Card): ScoreType {
```
Comment (zichun) [inline src/rules/tictactoepoker/tictactoepoker.ts:202]: don't have unit tests just for this function ah

File: src/rules/tictactoepoker/tictactoepoker.ts [hunk 2]
```diff
@@ -0,0 +1,752 @@
...
+                    return false;
+                }
+
+                return true;
+            })();
+
+
+            // Straight (just three consecutive cards)
+            const is_straight: boolean = (() => {
+                if (cards[0].get_value() + 1 == cards[1].get_value() &&
```
Comment (zichun) [inline src/rules/tictactoepoker/tictactoepoker.ts:232]: i'm surprise eslint didn't complain, but use === for strict equality (no implicit cast)

File: src/rules/tictactoepoker/tictactoepoker.ts [hunk 3]
```diff
@@ -0,0 +1,752 @@
...
+                for (let j = i + 1; j < cards.length; j++) {
+                    if (cards[i].get_value() == cards[j].get_value()) {
+                        return cards[i];
+                    }
+                }
+            }
+            return Card.EMPTY;
+        })();
+
+        if (pair_card != Card.EMPTY) {
```
Comment (zichun) [inline src/rules/tictactoepoker/tictactoepoker.ts:276]: same thing - !==

File: src/rules/tictactoepoker/tictactoepoker.ts [hunk 4]
```diff
@@ -0,0 +1,752 @@
...
+        }
+    };
+
+    // Since Aces can be 1 or 14, we have to try all combinations.
+    //
+    // Note the same Ace on the board can be treated differently
+    // for the different directions (rows, columns, diagonals)
+    let score = {combination_type: CombinationType.None};
+    let ace_state = [1, 1, 1];
+    for (ace_state[0] of [1, 14]) {
```
Comment (zichun) [inline src/rules/tictactoepoker/tictactoepoker.ts:300]: this enumerates all ace possibilities, even if there are less than 3 aces in the hand?

File: src/rules/tictactoepoker/tictactoepoker.ts (hunk 1 above)
Comment (chuanyu) [reply src/rules/tictactoepoker/tictactoepoker.ts:203]: No, this is tested indirectly via the tests on scoring on Board.

File: src/rules/tictactoepoker/tictactoepoker.ts (hunk 4 above)
Comment (chuanyu) [reply src/rules/tictactoepoker/tictactoepoker.ts]: Updated to only enumerate relevant possibilities.

File: src/rules/tictactoepoker/tictactoepoker.ts (hunk 3 above)
Comment (chuanyu) [reply src/rules/tictactoepoker/tictactoepoker.ts:274]: ESLint doesn't complain. What is being implicitly casted?

File: src/rules/tictactoepoker/tictactoepoker.ts (hunk 2 above)
Comment (chuanyu) [reply src/rules/tictactoepoker/tictactoepoker.ts]: ESLint doesn't complain. What is being implicitly casted?

Comment (zichun) [reply src/rules/tictactoepoker/tictactoepoker.ts]: nothing here, but it's just good practice. == will allow for implicit casting if types of operands mismatch, e.g the predicate (1 == true) evaluates to true.

**Pull request merged**